src/frequenz/sdk/microgrid/component/_component.py

```diff
@@ -106,13 +112,78 @@ def _component_category_from_protobuf(
     return ComponentCategory(component_category)
 
 
+class ComponentMetadata(ABC):
+    """Base class for component metadata classes."""
+
+    @property
+    @abstractmethod
+    def fuse(self) -> Optional[Fuse]:
+        """Get the fuse associated with this component.
+
+        Returns:
+            The fuse associated with this component.
+        """
+
+
+class GridMetadata(ComponentMetadata):
+    """Metadata for a grid connection point."""
+
+    def __init__(self, fuse: Fuse) -> None:
+        """Create a new instance.
+
+        Args:
+            fuse: The fuse at the grid connection point.
+        """
+        self._fuse = fuse
+
+    @property
+    def fuse(self) -> Optional[Fuse]:
```

Comment:
Nitpick: *modern* Python uses `Fuse | None` (but we are trying to modernize the code base, so it would be nice if new code uses the new style, although if the file in general is using the old style it is OK to leave it and make the modernization of the whole file in another commit/PR)

```suggestion
    def fuse(self) -> Fuse | None:
```

Reply:
Amazing! I inserted a commit to modernize the declarations, and then changed my commits to use this style.